apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/stu3/providers/TransformerUtils.java

```diff
@@ -2785,51 +2785,71 @@ public static String retrieveNpiCodeDisplay(String npiCode) {
+   * Reads ALL the NPI codes and display values from the NPI_Coded_Display_Values_Tab.txt file and,
+   * if present, the Synthea NPI file. Refer to the README file in the src/main/resources directory
    */
-  private static Map<String, String> readNpiCodeFile() {
-
-    Map<String, String> npiCodeMap = new HashMap<String, String>();
+  private static Map<String, String> readNpiCodeFiles() {
+    Map<String, String> npiCodeMap = new HashMap<>();
     try (final InputStream npiCodeDisplayStream =
             Thread.currentThread()
                 .getContextClassLoader()
                 .getResourceAsStream("NPI_Coded_Display_Values_Tab.txt");
         final BufferedReader npiCodesIn =
             new BufferedReader(new InputStreamReader(npiCodeDisplayStream))) {
-      /*
-       * We want to extract the NPI codes and display values and put in a map for easy retrieval to
-       * get the display value-- npiColumns[0] is the NPI Code, npiColumns[4] is the NPI
-       * Organization Code, npiColumns[8] is the NPI provider name prefix, npiColumns[6] is the NPI
-       * provider first name, npiColumns[7] is the NPI provider middle name, npiColumns[5] is the
-       * NPI provider last name, npiColumns[9] is the NPI provider suffix name, npiColumns[10] is
-       * the NPI provider credential.
-       */
-      String line = "";
-      npiCodesIn.readLine();
-      while ((line = npiCodesIn.readLine()) != null) {
-        String npiColumns[] = line.split("\t");
-        if (npiColumns[4].isEmpty()) {
-          String npiDisplayName =
-              npiColumns[8].trim()
-                  + " "
-                  + npiColumns[6].trim()
-                  + " "
-                  + npiColumns[7].trim()
-                  + " "
-                  + npiColumns[5].trim()
-                  + " "
-                  + npiColumns[9].trim()
-                  + " "
-                  + npiColumns[10].trim();
-          npiCodeMap.put(npiColumns[0], npiDisplayName.replace("  ", " ").trim());
-        } else {
-          npiCodeMap.put(npiColumns[0], npiColumns[4].replace("\"", "").trim());
-        }
-      }
+      npiCodeMap.putAll(readNpiCodeFile(npiCodesIn));
     } catch (IOException e) {
       throw new UncheckedIOException("Unable to read NPI code data.", e);
     }
+    // Merge Synthea generated NPIs if present. This block is commented out pending discussions
+    // on how to proceed with synthetic vs real NPIs.
+    /*
+    try {
+      URL syntheaNPIFile =
+          gov.cms.bfd.model.rif.samples.StaticRifResource.SYNTHEA_NPIS.getResourceUrl();
+      BufferedReader syntheaNpiCodesIn =
+          new BufferedReader(new InputStreamReader(syntheaNPIFile.openStream()));
+      Map<String, String> syntheaNpiCodes = readNpiCodeFile(syntheaNpiCodesIn);
+      npiCodeMap.putAll(syntheaNpiCodes);
+    } catch (IOException e) {
+      // Ignore, Synthea may not be used here
+      LOGGER.info("SYNTHEA NPI codes were not loaded");
+    }
+    */
```

Comment:
I think I'd prefer to **not** support lookups for the synthetic NPIs -- our end users won't be able to look them up, either, so pretending that we can feels like it might be even more confusing for them.

Thoughts?

Reply:
This was reverted, since we've decided to pull the NPI data changes out.